### PR TITLE
Adding Metadata Application

### PR DESCRIPTION
**Pull request by kegor**

### CODE REVIEW

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/client/apps/metadata-e2e/src/plugins/index.js  1
- src/Products/Metadata/Util/Directory/DirectoryUtils.cs  4
- src/Products/Common/Entity/Web/FilePropertyEntity.cs  1
- src/Products/Metadata/Entity/FilePropertyName.cs  1
- GroupDocs.Total.MVC.Test/MetadataControllerTest.cs  1
- src/client/apps/metadata-e2e/src/support/app.po.ts  1
- src/Products/Common/Util/Comparator/CaseInsensitiveComparer.cs  1
- src/Products/Metadata/Config/MetadataConfiguration.cs  3
- src/Products/Metadata/Controllers/MetadataApiController.cs  18
- src/Controllers/MetadataController.cs  1
         

Clones added
============
- GroupDocs.Total.MVC.Test/MetadataControllerTest.cs  7
- src/Products/Metadata/Controllers/MetadataApiController.cs  20
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085813&bid=16732229)

File: src/client/apps/metadata/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  name: 'metadata',
+  preset: '../../jest.config.js',
+  coverageDirectory: '../../coverage/apps/metadata',
+  snapshotSerializers: [
+    'jest-preset-angular/AngularSnapshotSerializer.js',
```
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           

Complexity increasing per file
==============================
- src/Products/Common/Util/Comparator/CaseInsensitiveComparer.cs  1
- GroupDocs.Total.MVC.Test/MetadataControllerTest.cs  1
- src/Controllers/MetadataController.cs  1
- src/Products/Common/Entity/Web/FilePropertyEntity.cs  1
- src/Products/Metadata/Util/Directory/DirectoryUtils.cs  4
- src/Products/Metadata/Entity/FilePropertyName.cs  1
- src/Products/Metadata/Controllers/MetadataApiController.cs  18
- src/client/apps/metadata-e2e/src/support/app.po.ts  1
- src/Products/Metadata/Config/MetadataConfiguration.cs  3
- src/client/apps/metadata-e2e/src/plugins/index.js  1
         

Clones added
============
- GroupDocs.Total.MVC.Test/MetadataControllerTest.cs  7
- src/Products/Metadata/Controllers/MetadataApiController.cs  20
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814&bid=16732230)

File: src/client/apps/metadata-e2e/src/support/app.po.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export const getGreeting = () => cy.get('h1');
```
Comment (bobkovalex) [inline src/client/apps/metadata-e2e/src/support/app.po.ts]: Codacy Issue found: ['cy' is not defined.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata-e2e/src/plugins/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+// the plugins file with the 'pluginsFile' configuration option.
+//
+// You can read more here:
+// https://on.cypress.io/plugins-guide
+// ***********************************************************
+
+// This function is called when a project is opened or re-opened (e.g. due to
+// the project's config changing)
+
+const { preprocessTypescript } = require('@nrwl/cypress/plugins/preprocessor');
```
Comment (bobkovalex) [inline src/client/apps/metadata-e2e/src/plugins/index.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata/jest.config.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  name: 'metadata',
+  preset: '../../jest.config.js',
+  coverageDirectory: '../../coverage/apps/metadata',
```
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata/jest.config.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  name: 'metadata',
+  preset: '../../jest.config.js',
```
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata/jest.config.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  name: 'metadata',
```
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/Controllers/MetadataController.cs [hunk 7]
```diff
@@ -0,0 +1,15 @@
+﻿using System.Web.Mvc;
+
+namespace GroupDocs.Total.MVC.Controllers
+{
+    /// <summary>
+    /// Metadata Web page controller
+    /// </summary>
+    public class MetadataController : Controller
+    {       
+        public ActionResult Index()
```
Comment (bobkovalex) [inline src/Controllers/MetadataController.cs]: Codacy Issue found: [Make 'Index' a static method.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata/jest.config.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+  preset: '../../jest.config.js',
+  coverageDirectory: '../../coverage/apps/metadata',
+  snapshotSerializers: [
+    'jest-preset-angular/AngularSnapshotSerializer.js',
+    'jest-preset-angular/HTMLCommentSerializer.js'
```
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata-e2e/src/support/app.po.ts (hunk 2 above)
Comment (bobkovalex) [inline src/client/apps/metadata-e2e/src/support/app.po.ts]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata-e2e/src/plugins/index.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+module.exports = (on, config) => {
+  // `on` is used to hook into various events Cypress emits
+  // `config` is the resolved Cypress config
+
+  // Preprocess Typescript
+  on('file:preprocessor', preprocessTypescript(config));
```
Comment (bobkovalex) [inline src/client/apps/metadata-e2e/src/plugins/index.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           

Complexity increasing per file
==============================
- src/Products/Metadata/Util/Directory/DirectoryUtils.cs  4
- GroupDocs.Total.MVC.Test/MetadataControllerTest.cs  1
- src/client/apps/metadata-e2e/src/plugins/index.js  1
- src/Products/Metadata/Controllers/MetadataApiController.cs  18
- src/Products/Metadata/Config/MetadataConfiguration.cs  3
- src/Controllers/MetadataController.cs  1
- src/Products/Common/Util/Comparator/CaseInsensitiveComparer.cs  1
- src/client/apps/metadata-e2e/src/support/app.po.ts  1
- src/Products/Metadata/Entity/FilePropertyName.cs  1
- src/Products/Common/Entity/Web/FilePropertyEntity.cs  1
         

Clones added
============
- GroupDocs.Total.MVC.Test/MetadataControllerTest.cs  7
- src/Products/Metadata/Controllers/MetadataApiController.cs  20
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814&bid=16732230)

File: src/client/apps/metadata-e2e/src/plugins/index.js (hunk 9 above)
Comment (bobkovalex) [inline src/client/apps/metadata-e2e/src/plugins/index.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/Controllers/MetadataController.cs (hunk 7 above)
Comment (bobkovalex) [inline src/Controllers/MetadataController.cs]: Codacy Issue found: [Make 'Index' a static method.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata-e2e/src/support/app.po.ts (hunk 2 above)
Comment (bobkovalex) [inline src/client/apps/metadata-e2e/src/support/app.po.ts]: Codacy Issue found: ['cy' is not defined.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata/jest.config.js (hunk 6 above)
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata/jest.config.js (hunk 4 above)
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata-e2e/src/plugins/index.js (hunk 3 above)
Comment (bobkovalex) [inline src/client/apps/metadata-e2e/src/plugins/index.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata/jest.config.js (hunk 8 above)
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata/jest.config.js (hunk 1 above)
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata/jest.config.js (hunk 5 above)
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata-e2e/src/support/app.po.ts (hunk 2 above)
Comment (bobkovalex) [inline src/client/apps/metadata-e2e/src/support/app.po.ts]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Metadata/Entity/FilePropertyName.cs  1
- src/Controllers/MetadataController.cs  1
- src/client/apps/metadata-e2e/src/support/app.po.ts  1
- GroupDocs.Total.MVC.Test/MetadataControllerTest.cs  1
- src/client/apps/metadata-e2e/src/plugins/index.js  1
- src/Products/Metadata/Controllers/MetadataApiController.cs  18
- src/Products/Metadata/Config/MetadataConfiguration.cs  3
- src/Products/Common/Util/Comparator/CaseInsensitiveComparer.cs  1
- src/Products/Metadata/Util/Directory/DirectoryUtils.cs  4
- src/Products/Common/Entity/Web/FilePropertyEntity.cs  1
         

Clones added
============
- GroupDocs.Total.MVC.Test/MetadataControllerTest.cs  7
- src/Products/Metadata/Controllers/MetadataApiController.cs  20
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085813&bid=16732229)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 10
           

Complexity increasing per file
==============================
- src/Controllers/MetadataController.cs  1
- src/client/apps/metadata-e2e/src/support/app.po.ts  1
- src/Products/Metadata/Entity/FilePropertyName.cs  1
- src/Products/Metadata/Util/Directory/DirectoryUtils.cs  4
- GroupDocs.Total.MVC.Test/MetadataControllerTest.cs  1
- src/Products/Metadata/Controllers/MetadataApiController.cs  18
- src/Products/Metadata/Config/MetadataConfiguration.cs  3
- src/Products/Common/Util/Comparator/CaseInsensitiveComparer.cs  1
- src/Products/Common/Entity/Web/FilePropertyEntity.cs  1
- src/client/apps/metadata-e2e/src/plugins/index.js  1
         

Clones added
============
- GroupDocs.Total.MVC.Test/MetadataControllerTest.cs  6
- src/Products/Metadata/Controllers/MetadataApiController.cs  20
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814&bid=16732230)

File: src/client/apps/metadata-e2e/src/plugins/index.js (hunk 9 above)
Comment (bobkovalex) [inline src/client/apps/metadata-e2e/src/plugins/index.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/Controllers/MetadataController.cs (hunk 7 above)
Comment (bobkovalex) [inline src/Controllers/MetadataController.cs]: Codacy Issue found: [Make 'Index' a static method.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata/jest.config.js (hunk 6 above)
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata/jest.config.js (hunk 1 above)
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata-e2e/src/plugins/index.js (hunk 3 above)
Comment (bobkovalex) [inline src/client/apps/metadata-e2e/src/plugins/index.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata/jest.config.js (hunk 4 above)
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata/jest.config.js (hunk 5 above)
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata/jest.config.js (hunk 8 above)
Comment (bobkovalex) [inline src/client/apps/metadata/jest.config.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/client/apps/metadata-e2e/src/support/app.po.ts (hunk 2 above)
Comment (bobkovalex) [inline src/client/apps/metadata-e2e/src/support/app.po.ts]: Codacy Issue found: ['cy' is not defined.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

Comment (bobkovalex) [inline src/client/apps/metadata-e2e/src/support/app.po.ts]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/Products/Metadata/Util/Directory/DirectoryUtils.cs  4
- src/Products/Metadata/Config/MetadataConfiguration.cs  3
- GroupDocs.Total.MVC.Test/MetadataControllerTest.cs  1
- src/client/apps/metadata-e2e/src/support/app.po.ts  1
- src/Products/Common/Entity/Web/FilePropertyEntity.cs  1
- src/Products/Common/Util/Comparator/CaseInsensitiveComparer.cs  1
- src/Products/Metadata/Controllers/MetadataApiController.cs  18
- src/Controllers/MetadataController.cs  1
- src/Products/Metadata/Entity/FilePropertyName.cs  1
- src/client/apps/metadata-e2e/src/plugins/index.js  1
         

Clones added
============
- GroupDocs.Total.MVC.Test/MetadataControllerTest.cs  6
- src/Products/Metadata/Controllers/MetadataApiController.cs  20
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085813&bid=16732229)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/Controllers/MetadataController.cs  1
- src/Products/Metadata/Controllers/MetadataApiController.cs  18
- src/Products/Common/Util/Comparator/CaseInsensitiveComparer.cs  1
- src/Products/Metadata/Config/MetadataConfiguration.cs  3
- src/Products/Metadata/Entity/FilePropertyName.cs  1
- src/Products/Common/Entity/Web/FilePropertyEntity.cs  1
- src/Products/Metadata/Util/Directory/DirectoryUtils.cs  4
- src/client/apps/metadata-e2e/src/plugins/index.js  1
- GroupDocs.Total.MVC.Test/MetadataControllerTest.cs  1
- src/client/apps/metadata-e2e/src/support/app.po.ts  1
         

Clones added
============
- src/Products/Metadata/Controllers/MetadataApiController.cs  20
- GroupDocs.Total.MVC.Test/MetadataControllerTest.cs  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814&bid=16732230)

File: src/client/apps/metadata-e2e/src/support/app.po.ts (hunk 2 above)
Comment (bobkovalex) [inline src/client/apps/metadata-e2e/src/support/app.po.ts]: Codacy Issue found: ['cy' is not defined.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

File: src/Controllers/MetadataController.cs (hunk 7 above)
Comment (bobkovalex) [inline src/Controllers/MetadataController.cs]: Codacy Issue found: [Make 'Index' a static method.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5085814)

**Pull request merged**